full-moon/src/ast/parsers.rs

```diff
@@ -1435,6 +1402,33 @@ cfg_if::cfg_if! {
                             return_type: Box::new(return_value),
                         },
                     )
+                } else {
+                    // Tuples are only permitted as the return type of a function
+                    // However, if we just have a single type wrapped around in parentheses, its not really a tuple - this is allowed anywhere.
+
+                    match this.0 {
+                        TypeInfoContext::ReturnType => {},
+                        _ => {
+                            let num_types_present = types.len();
+                            if num_types_present != 1 {
+                                return Err(InternalAstError::UnexpectedToken {
+                                    token: match num_types_present {
+                                        0 => end_parenthese,
+                                        _ => types.pairs().next().unwrap().punctuation().unwrap().clone(), // A comma after the first type must exist if > 1 types present
```

Comment:
What the heck is this doing?

Reply:
Can you use `expect` both to document what's actually going on and in case this fails, we'll know what it is?

Reply:
This is a weird one. If we reach this match, it means there was either 0 or 1+ types inside the parentheses, which are only possible as a function return type. I had it simpler before the last commit, but I had to do this so that the error would point to the right token, in this case the comma after the first type.

I don't think `except` is possible here - we have already parsed the tokens before we reached here, now we are checking what we have parsed to see if it's valid. Im busy for the next 2 weeks, but I can look into this further to make it more understandable. (Feel free to also change it yourself if you want)

Reply:
`expect` as in, instead of `unwrap` to document what went wrong if they failed.